Tests/Templating/TemplateGuesserTest.php

```diff
@@ -103,6 +103,44 @@ public function testGuessTemplateWithoutBundle()
         $this->assertEquals(':OutOfBundle:index.html.twig', (string) $templateReference);
     }
 
+    public function testGuessTemplateWithInvokeMagicMethod()
+    {
+        $templateGuesser = new TemplateGuesser($this->kernel);
+
+        $templateReference = $templateGuesser->guessTemplateName(array(
+            new Fixture\FooBundle\Controller\FooController(),
+            '__invoke',
+        ), new Request());
+
+        $this->assertEquals('FooBundle::Foo.html.twig', (string) $templateReference);
+    }
+
+    public function testGuessTemplateWithACustomPattern()
+    {
+        $templateGuesser = new TemplateGuesser($this->kernel, array('/foobar/', '/FooBundle\\\Action\\\(.+)Action/'));
+
+        $templateReference = $templateGuesser->guessTemplateName(array(
+            new Fixture\FooBundle\Action\FooAction(),
+            'indexAction',
```

Comment:
Ok got it.

The is few more cases:

* Implicit reference to `__invoke`: https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpKernel/Controller/ControllerResolver.php#L60-L63
* Action method not called `__invoke` and not finishing by `Action`

Ex:

```php
class MyAction()
{
    /** @Template */
    public foo() /* no Action */
    {
    }

    /** @Template */
    public function bar()
    {
    }
}